### PR TITLE
Feature/improved mushdoom

### PR DESCRIPTION
**Pull request by JoanStinson**
What's new:

- Basically improved combat and feedback of enemies (bigger, new run and hit animations, particle effects, fixes, horizontal throw, etc.)

### CODE REVIEW

File: Scripts/src/Script/EnemyController.cpp [hunk 1]
```diff
@@ -145,20 +145,27 @@ void EnemyController::Seek(const float3& velocity) const
...
-
 	if (App->artificial_intelligence->IsPointWalkable(position) || grounded)
 	{
-			float3 look_pos = current_target->owner->transform.GetGlobalTranslation();
-			look_pos.y = owner->transform.GetGlobalTranslation().y;
-			owner->transform.LookAt(look_pos);
-			owner->transform.SetGlobalMatrixTranslation(position);
+		//float3 look_pos = current_target->owner->transform.GetGlobalTranslation();
+		//look_pos.y = owner->transform.GetGlobalTranslation().y;
+		//owner->transform.LookAt(look_pos);
```
Comment (Dartemiss) [inline Scripts/src/Script/EnemyController.cpp:159]: Why is this code commented?

File: Scripts/src/Script/EnemyController.cpp [hunk 2]
```diff
@@ -145,20 +145,27 @@ void EnemyController::Seek(const float3& velocity) const
...
+
+		float smooth_rotation = 0.001f;
+		float3 desired_direction = current_target->owner->transform.GetGlobalTranslation() - owner->transform.GetGlobalTranslation();
+		desired_direction.y = 0;
+		Quat desired_rotation = Quat::LookAt(float3::unitZ, desired_direction.Normalized(), float3::unitY, float3::unitY);
+		Quat new_rotation = Quat::Slerp(owner->transform.GetGlobalRotation(), desired_rotation, App->time->delta_time * smooth_rotation);
+		owner->transform.SetGlobalMatrixRotation(new_rotation);
+
+		owner->transform.SetGlobalMatrixTranslation(position);
```
Comment (Dartemiss) [inline Scripts/src/Script/EnemyController.cpp:168]: I suppose this is what provoke the bug of michael jackson. It should always rotate forward not backwards.

File: Scripts/src/Script/Mushdoom.cpp [hunk 3]
```diff
@@ -285,7 +312,7 @@ void Mushdoom::BattleCircleAI()
 
 void Mushdoom::PursueThrowAI()
 {
-//	if (!IsGrounded() && move_with_physics) return;
+	//	if (!IsGrounded() && move_with_physics) return;
```
Comment (Dartemiss) [inline Scripts/src/Script/Mushdoom.cpp:315]: Erase if useless

File: Scripts/src/Script/Mushdoom.cpp [hunk 4]
```diff
@@ -307,7 +334,36 @@ void Mushdoom::PursueThrowAI()
...
+
+		//if (!desired_velocity.Equals(float3::zero))
+		//{
+		//	Avoid(desired_velocity);
+		//}
+		//else
+		//{
+		//	desired_velocity = target - position;
+		//	Seek(desired_velocity);
+		//}
```
Comment (Dartemiss) [inline Scripts/src/Script/Mushdoom.cpp:351]: if useless erase it

File: Scripts/src/Script/Mushdoom.cpp [hunk 5]
```diff
@@ -319,20 +375,14 @@ void Mushdoom::PursueThrowAI()
...
-	else
-	{
-		is_moving = false;
-	}
 }
 
 void Mushdoom::IdleThrowAI()
 {
-//	if (!IsGrounded() && move_with_physics) return;
+	//	if (!IsGrounded() && move_with_physics) return;
```
Comment (Dartemiss) [inline Scripts/src/Script/Mushdoom.cpp:385]: Erased it if useless

File: Scripts/src/Script/MushdoomStateGround.cpp [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+	}
+}
+
+void MushdoomStateGround::OnStateUpdate()
+{
+}
+
+void MushdoomStateGround::OnStateExit()
+{
+}
```
Comment (Dartemiss) [inline Scripts/src/Script/MushdoomStateGround.cpp:31]: When is this state used now?

File: Scripts/src/Script/PlayerController.cpp [hunk 7]
```diff
@@ -111,112 +111,112 @@ void PlayerController::Update()
...
-						current_dash_time -= App->time->delta_time;
-					}
-					else
-					{
-						is_dashing = false;
-						current_dash_cooldown = dash_cooldown;
-					}
-
-					return;
+					player_movement->MoveIntoDirection(owner->transform.GetFrontVector(), dash_speed);
```
Comment (Dartemiss) [inline Scripts/src/Script/PlayerController.cpp:123]: Why are you changing dash stuff?

File: Scripts/src/Script/EnemyController.cpp (hunk 1 above)
Comment (JoanStinson) [reply Scripts/src/Script/EnemyController.cpp:159]: old way the enemies rotated will delete

File: Scripts/src/Script/EnemyController.cpp (hunk 2 above)
Comment (JoanStinson) [reply Scripts/src/Script/EnemyController.cpp]: resolved

File: Scripts/src/Script/Mushdoom.cpp (hunk 3 above)
Comment (JoanStinson) [reply Scripts/src/Script/Mushdoom.cpp:315]: done

File: Scripts/src/Script/Mushdoom.cpp (hunk 4 above)
Comment (JoanStinson) [reply Scripts/src/Script/Mushdoom.cpp:351]: done

File: Scripts/src/Script/Mushdoom.cpp (hunk 5 above)
Comment (JoanStinson) [reply Scripts/src/Script/Mushdoom.cpp:385]: done

File: Scripts/src/Script/MushdoomStateGround.cpp (hunk 6 above)
Comment (JoanStinson) [reply Scripts/src/Script/MushdoomStateGround.cpp:31]: It's not used now, will be used when mushdooms come from the floor

File: Scripts/src/Script/PlayerController.cpp (hunk 7 above)
Comment (JoanStinson) [reply Scripts/src/Script/PlayerController.cpp:123]: sorry

**Pull request merged**